app/src/components/buttons/NavbarItemLinkButton.tsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import styled from '@emotion/styled';
+import { Button } from 'src/components/base';
+import { pxToRem, colors } from 'src/styled-theme';
+
+export interface NavbarItemLinkButtonProps {
+  readonly title: string;
+  readonly children: React.ReactNode;
+
+  readonly selectedRoute: any;
```

Comment:
@DorianKwan @mgibson-casper 

This is the button itself for: 

  const [selectedRoute, setSelectedRoute] = useState(null);

  const handleNavItemSelection = (event: {
    target: { innerText: React.SetStateAction<null> };
  }) => {
    setSelectedRoute(event.target.innerText);
  };

Reply:
I really don't think we need to type this as `any`. The `selectedRoute` should very likely be the route object, something like:

```
{
  title: string,
  id: number
  ...
}
```

Reply:
Or I guess it could be a `string`, but then we should call the param something like `selectedRouteText`.

Reply:
Yeah, I've just updated all to `<string>` and stuff seems to work fine.  Now to fix failing tests!